x-element.js

```diff
@@ -1294,100 +1226,113 @@ class Template {
       }
       // Special case to handle elements which only allow text content (no comments).
       const localName = node.localName;
-      if ((localName === 'style' || localName === 'script') && Template.#CONTENT_REGEX.exec(node.textContent)) {
+      if (
+        (localName === 'style' || localName === 'script') &&
+        TemplateEngine.#CONTENT_REGEX.exec(node.textContent)
+      ) {
         throw new Error(`Interpolation of "${localName}" tags is not allowed.`);
-      } else if (localName === 'plaintext' || localName === 'textarea' || localName === 'title') {
-        const contentMatch = Template.#CONTENT_REGEX.exec(node.textContent);
+      } else if (
+        localName === 'plaintext' ||
+        localName === 'textarea' ||
+        localName === 'title'
+      ) {
+        const contentMatch = TemplateEngine.#CONTENT_REGEX.exec(node.textContent);
         if (contentMatch) {
-          items.push({ path, key: contentMatch[1], type: 'text' });
+          initialDirections.push({ path, key: contentMatch[1], type: 'text' });
         }
       }
     } else if (node.nodeType === Node.COMMENT_NODE) {
-      const contentMatch = Template.#CONTENT_REGEX.exec(node.textContent);
+      const contentMatch = TemplateEngine.#CONTENT_REGEX.exec(node.textContent);
       if (contentMatch) {
         node.textContent = '';
         const startNode = document.createComment('');
         node.parentNode.insertBefore(startNode, node);
         path[path.length - 1] = path[path.length - 1] + 1;
-        items.push({ path, key: contentMatch[1], type: 'content' });
+        initialDirections.push({ path, key: contentMatch[1], type: 'content' });
       }
     }
     let iii = 0;
     for (const childNode of node.childNodes) {
-      items.push(...Template.#evaluate(childNode, [...path, iii++]));
+      initialDirections.push(...TemplateEngine.#getInitialDirections(childNode, [...path, iii++]));
+    }
+    return initialDirections;
+  }
+
+  static #fillInHtml(type, strings) {
```

Comment:
`populateHTML` ?

Reply:
Going to leave this as-is in here for now.